stages/org.osbuild.grub2

```diff
@@ -132,11 +142,77 @@ STAGE_OPTS = """
     "description": "Whether to write /etc/defaults/grub",
     "type": "boolean",
     "default": true
+  },
+  "ignition": {
+    "description": "Include ignition support in the grub.cfg",
+    "type": "boolean",
+    "default": false
   }
 }
 """
 
 
+# The main grub2 configuration file template. Used for UEFI and legacy
+# boot. The parameters are currently:
+#   - $search: to specify the search criteria of how to locate grub's
+#     "root device", i.e. the device where the "OS images" are stored.
+#   - $ignition: configuration for ignition, if support for ignition
+#     is enabled
+GRUB_CFG_TEMPALTE = """
+set timeout=0
+load_env
+search --no-floppy --set=root $search
+set boot=$${root}
+function load_video {
+  insmod all_video
+}
+${ignition}
+blscfg
+"""
+
+
+# The grub2 redirect configuration template. This is used in case of
+# hybrid (uefi + legacy) boot. In this case this configuration, which
+# is located in the EFI directory, will redirect to the main grub.cfg
+# (GRUB_CFG_TEMPLATE).
+# The parameters are:
+#   - $root: specifies the path to the grub2 directory relative to
+#     to the file-system where the directory is located on
+GRUB_REDIRECT_TEMPALTE = """
+search --no-floppy --set prefix --file ${root}grub2/grub.cfg
+set prefix=($$prefix)${root}grub2
+configfile $$prefix/grub.cfg
+"""
+
+
+# Template for ignition support in the grub.cfg
+#
+# it was taken verbatim from Fedora CoreOS assembler's grub.cfg
+# See https://github.com/coreos/coreos-assembler/
+#
+# The parameters are:
+#   - $root: specifies the path to the grub2 directory relative
+#     to the file-system where the directory is located on
+
+IGNITION_TEMPLATE = """
+# Ignition support
+set ignition_firstboot=""
+if [ -f "${root}ignition.firstboot" ]; then
+    # default to dhcp networking parameters to be used with ignition
+    set ignition_network_kcmdline='rd.neednet=1 ip=dhcp,dhcp6'
```

Comment:
We're already moving away from this with FCOS - see also https://github.com/coreos/coreos-assembler/pull/1373

Reply:
Good to know! Thanks for the heads up.

Reply:
For reference, those are moving to here: https://github.com/coreos/fedora-coreos-config/pull/427